app/src/main/java/br/org/cesar/discordtime/stickysessions/navigation/wrapper/ViewStarter.java

```diff
@@ -22,17 +23,24 @@
 
     public void goNext(Context context, String activityName, boolean shouldClearStack)
             throws InvalidViewNameException{
-        goNext(context, activityName, shouldClearStack, new Bundle());
+        goNext(context, activityName, shouldClearStack, null);
     }
 
     public void goNext(Context context, String activityName, boolean shouldClearStack,
-                       Bundle extras) throws InvalidViewNameException{
+                       IBundle iBundle) throws InvalidViewNameException{
         Class<? extends Activity> className = activities.get(activityName);
         if(className == null){
             throw new InvalidViewNameException(activityName);
         }
         Intent intent = new Intent(context, className);
-        intent.putExtras(extras);
+
+        if (iBundle != null) {
+            Bundle bundle = new Bundle();
+            for (Map.Entry<String, String> entry : iBundle.getExtras().entrySet()) {
+                bundle.putString(entry.getKey(), entry.getValue());
```

Comment:
Shouldn't there be a method in bundle wrapper that returns a bundle? Why interact every value and adding it to a bundle?

Reply:
Using the iBundle this way (in the current code) we could not add elements other than "String".

Reply:
@ehammo If we put android.os.Bundle as the return type of one of the methods from the interface it is going to contaminate Presenter with android dependencies.

@ericbbraga yeah, I know. I kept it like that because It is the only type we're using right now. I'm not sure if worths implementing all signatures for all types or use HashMap of Objects and deal with all cast situations.